luthier/config/src/utils/tableUtils.lua

```diff
@@ -12,33 +12,72 @@ local misc = require 'utils./misc'
 -- Build Config
 -------------------------------------------------------------------------------
 
+--- Build a specific physical table into a json object.
+--
+-- for each of the pair in logicalToPhysicalColumnNames, e.g.
+--
+-- logicalToPhysicalColumnNames = {
+--     ...
+--     testDimension = "testDimensionPhysicalName",
+--     ...
+-- }
+--
+-- the corresponding Json will be
+--
+-- "logicalToPhysicalColumnNames": [
+--   ...
+--   {
+--     "logicalName": "testDimension"
+--     "physicalName": "testDimensionPhysicalName",
+--   },
+--   ...
+-- ]
+--
+-- this reformatting is done to avoid a gotcha since we don't want the Json
+-- object to stay as an ArrayNode. If we instead naively translate this, then it
+-- will be an ObjectNode, i.e. { "logiName": "physiName" }, when it is non-empty,
+-- but an ArrayNode, i.e. [], when it is empty.
+local physical_table_build = function(name, physical_table)
+    local copy = misc.shallow_copy(physical_table)
+    copy.type = copy.type or "strict"
+    copy.description = copy.description or name
+    copy.physicalTables = copy.physicalTables or {}
+    copy.dateTimeZone = copy.dateTimeZone or "UTC"
+    local intermediate_map = copy.logicalToPhysicalColumnNames or {}
+    -- rebuilds the logicalToPhysicalColumnNames map with nicer format
+    copy.logicalToPhysicalColumnNames = {}
+    for logical_name, physical_name in pairs(intermediate_map) do
+        local name_pair = {
+            logicalName = logical_name,
+            physicalName = physical_name
+        }
+        table.insert(copy.logicalToPhysicalColumnNames, name_pair)
+    end
+    return copy
+end
+
 --- Add physical table configs and logical table configs into a configuration.
 --
--- @param tables A table containing two keys: 
+-- @param tables A table containing two keys:
 --  physical - A table of physical table configuration, keyed on name
 --  logical - A table of logical table configuration, keyed on name
 -- @return The table for storing table configs and ready be parsed into json
 function M.build_table_config(tables)
 
-    local configuration = {
-        physical = {},
-        logical = {}
-    }
+    local physical = {}
+    local logical = {}
 
     for name, physical_table in pairs(tables.physical) do
-        local copy = misc.shallow_copy(physical_table)
-        copy.name = copy.name or name
-        copy.description = copy.description or name
-        table.insert(configuration.physical, copy)
+        physical[name] = physical_table_build(name, physical_table)
```

Comment:
We should be doing `physical[copy.name or name]` here like we do with `logical`.

Reply:
I don't think we should. For one, the table is not doing that. 

For table, dimension, and metric, I think we had the discussion to transform the lua format from 
```
"objName": {
    "name": "objName",
    "property1": "objProperty1",
    ...
}
```
```
"objName": {
    "property1": "objProperty1",
    ...
}
```
Since the prior one does not give any benefit for the user. Since we have made the examples in `dimensions.lua` to look like the later one, there should not be this defaulting in `*Utils.lua` to hint otherwise.

Reply:
We should update the documentation around tables then, because I think the documentation still refers to a `name` field that defaults to the key.

Reply:
Yeah, that is part of my change that will be pushed in a couple of minutes.